test/behaviour/concept/thing/entity/EntitySteps.jl

```diff
@@ -1,85 +1,216 @@
+    @expect res === true
+end
+
+@then("entity(person) get instances is empty") do context
+    attr = g.get(context[:cm], EntityType, "person")
+    res_ent = g.get_instances(g.as_remote(attr, context[:transaction]))
+    @expect isempty(res_ent) === true
+end
+
+# Scenario: Entity can have keys
+@when("\$alice = attribute(username) as(string) put: alice") do context
+    attr_type = g.get(context[:cm], AttributeType, "username")
+    res = g.put(g.as_remote(attr_type, context[:transaction]), "alice")
+    context[:alice] = res
+end
+
+@when("entity \$a set has: \$alice") do context
+    set_has(context[:transaction], context[:a], context[:alice])
+end
+
+@then("entity \$a get attributes(username) as(string) contain: \$alice") do context
+    attr = get(ConceptManager(context[:transaction]), AttributeType, "username")
+    inst_usernames = get_has(context[:transaction], context[:a], attr)
+    @expect in(context[:alice], inst_usernames)
+end
+
+@then("entity \$a get keys contain: \$alice") do context
+    inst_keys = get_has(context[:transaction], context[:a] , nothing, nothing, true)
+    @expect in(context[:alice], inst_keys)
+end
+
+@then("attribute \$alice get owners contain: \$a") do context
+    owners = get_owners(context[:transaction], context[:alice])
+    @expect in(context[:a], owners)
+end
+
+@when("\$alice = attribute(username) as(string) get: alice") do context
+    attr = get(ConceptManager(context[:transaction]), AttributeType, "username")
+    res = g.get_instances(g.as_remote(attr, context[:transaction]))
+    context[:alice] = res[1]
+end
+
+# Scenario: Entity can unset keys
+@when("entity \$a unset has: \$alice") do context
+    g.unset_has(context[:transaction], context[:a], context[:alice])
+end
+
+@then("entity \$a get attributes(username) as(string) do not contain: \$alice") do context
+    attr = get(ConceptManager(context[:transaction]), AttributeType, "username")
+    inst_usernames = get_has(context[:transaction], context[:a], attr)
+    @expect !in(context[:alice], inst_usernames)
+end
+
+@then("entity \$a get keys do not contain: \$alice") do context
+    inst_keys = get_has(context[:transaction], context[:a] , nothing, nothing, true)
+    @expect !in(context[:alice], inst_keys)
+end
+
+@then("attribute \$alice get owners do not contain: \$a") do context
+    owners = get_owners(context[:transaction], context[:alice])
+    @expect !in(context[:a], owners)
+end
+
+# Scenario: Entity cannot have more than one key for a given key type
+@when("\$bob = attribute(username) as(string) put: bob") do context
+    attr_type = g.get(context[:cm], AttributeType, "username")
+    res = g.put(g.as_remote(attr_type, context[:transaction]), "bob")
+    context[:bob] = res
+end
+
+@then("entity \$a set has: \$bob; throws exception") do context
+    try
+        set_has(context[:transaction], context[:a], context[:bob])
+    catch ex
+        @expect ex !== nothing
+    end
```

Comment:
Do you care about the type/cause of exception?

Reply:
No. Here only the occurence of an exception counts.